extensions/json-language-features/server/README.md

```diff
@@ -143,34 +143,37 @@ In addition to the settings, schemas associations can also be provided through a
 
 Notification:
 - method: 'json/schemaAssociations'
-- params: `ISchemaAssociations | SchemaConfiguration[]` defined as follows
+- params: `ISchemaAssociations` or `ISchemaAssociation[]` defined as follows
 
 ```ts
-/**
- * An object where:
- *  - keys are file names or file paths (separated by `/`). `*` can be used as a wildcard.
- *  - values ar an arrays of schema URLs
- */
-type ISchemaAssociations = Record<string, string[]>
-
-interface SchemaConfiguration {
-    /**
-     * The URI of the schema, which is also the identifier of the schema.
-     */
-    uri: string;
-    /**
-     * A list of file path patterns that are associated to the schema. The '*' wildcard can be used. Exclusion patterns starting with '!'.
-     * For example '*.schema.json', 'package.json', '!foo*.schema.json'.
-     * A match succeeds when there is at least one pattern matching and last matching pattern does not start with '!'.
-     */
-    fileMatch?: string[];
-    /**
-     * The schema for the given URI.
-     * If no schema is provided, the schema will be fetched with the schema request service (if available).
-     */
-    schema?: JSONSchema;
+interface ISchemaAssociations {
+  /**
+   * An object where:
+   *  - keys are file names or file paths (using `/` as path separator). `*` can be used as a wildcard.
+   *  - values are an arrays of schema URIs
+   */
+  [pattern: string]: string[];
 }
+
+interface ISchemaAssociation {
+  /**
+   * The URI of the schema, which is also the identifier of the schema.
+   */
+  uri: string;
+
+  /**
+   * A list of file path patterns that are associated to the schema. The '*' wildcard can be used. Exclusion patterns starting with '!'.
+   * For example '*.schema.json', 'package.json', '!foo*.schema.json'.
+   * A match succeeds when there is at least one pattern matching and last matching pattern does not start with '!'.
+   */
+  fileMatch: string[];
+
+}
```

Comment:
`schema` property is also supported.
The type could be set to `object` if you didn't want to reference `JSONSchema`.

Reply:
Wait, I see what you did there now.

It's no longer accurate again after your change...

Reply:
This change was meant to describe two possible inputs for that notification:

1. An object form.
Example:
```js
{
  'foo.js': ['http://schemas.org/foo.json'],
  'bar.js': ['http://schemas.org/bar.json'],
}
```

2. An array form.
Example:
```js
[
  {
    fileMatch: ['foo.js'],
    uri: ['http://schemas.org/foo.json'],
  },
  {
    fileMatch: ['bar.js'],
    uri: ['http://schemas.org/bar.json'],
  },
]
```

With your change, we're back to just explaining the first form.

Reply:
No, sorry, was confused by `ISchemaAssociations`/`ISchemaAssociation[]` naming. Thought that `ISchemaAssociations` is just an array of `ISchemaAssociation` and thus equivalent.

I'm just wondering why `schema` property was removed from the interface documentation.